src/metrics/release_versions/getVersionByString.ts

```diff
@@ -0,0 +1,18 @@
+import { SemVer, clean as semverClean, parse as semverParse } from "semver";
+import { pick } from "ramda";
+import { ReleaseVersion } from "./interfaces";
+import { LogErrors } from "../../shared/logMessages";
+
+export const getVersionByString = (ref: string): ReleaseVersion => {
+  const version: SemVer | null = semverParse(semverClean(ref));
+  if (version === null) {
+    // this error should already be caught by the metric condition and type check of the metric itself
+    throw new Error(
+      `${LogErrors.releaseVersionsUnexpectedInvalidVersion}: ${ref}`
+    );
+  }
+  return pick(
+    ["raw", "major", "minor", "patch", "prerelease", "build", "version"],
+    version
+  );
```

Comment:
I think pick is unfortunately untyped and returns `any`, so if we want to make typescript compare the types we'd need to type this manually `as Pick<SemVer, "raw" | "major" | ...>`.

Maybe we can also add ramda typings from somewhere, but i'd expect them to cause a few issues in places.

Reply:
Good call!